jerry-core/parser/js/js-parser-expr.c

```diff
@@ -791,11 +791,21 @@ parser_parse_object_literal (parser_context_t *context_p) /**< context */
         uint32_t status_flags;
         cbc_ext_opcode_t opcode;
         uint16_t literal_index, function_literal_index;
+        bool is_getter = context_p->token.type == LEXER_PROPERTY_GETTER;
 #if !ENABLED (JERRY_ES2015)
         parser_object_literal_item_types_t item_type;
+#else /* ENABLED (JERRY_ES2015) */
+        if (lexer_check_next_character (context_p, LIT_CHAR_LEFT_PAREN))
+        {
+          lexer_construct_literal_object (context_p,
+                                          (is_getter ? (lexer_lit_location_t *) &lexer_get_literal
+                                                     : (lexer_lit_location_t *) &lexer_set_literal),
+                                          LEXER_STRING_LITERAL);
+          goto parse_object_method;
```

Comment:
Can we avoid this goto?

Reply:
Sure! I've reordered the switch cases to substitute the goto with a fallthru.